repair/row_level.cc

```diff
@@ -2499,6 +2501,25 @@ future<> repair_service::init_ms_handlers() {
     ms.register_repair_get_diff_algorithms([] (const rpc::client_info& cinfo) {
         return make_ready_future<std::vector<row_level_diff_detect_algorithm>>(suportted_diff_detect_algorithms());
     });
+    ms.register_repair_update_system_table([this] (const rpc::client_info& cinfo, repair_update_system_table_request req) -> future<repair_update_system_table_response> {
+        auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
+        // db_clock is ms. api::timestamp_clock is us
+        auto ts = api::timestamp_type(req.repair_time.time_since_epoch().count() * 1000);
+        rlogger.info("repair[{}]: Got repair_update_system_table_request from node={}, range={}, ts={}, repair_time={}", req.repair_uuid, from, req.range, ts, req.repair_time);
+        auto& db = this->get_db();
+        co_await db.invoke_on_all([&req, ts] (database& local_db) {
+            auto& table = local_db.find_column_family(req.table_uuid);
+            return table.update_repair_time(req.range, ts);
+        });
+        sstring cql = format("INSERT INTO system.{} (table_uuid, repair_time, repair_uuid, keyspace_name, table_name, range_start, range_end) VALUES (?, ?, ?, ?, ?, ?, ?)",
+                db::system_keyspace::REPAIR);
+        auto range_start = req.range.start() ? req.range.start()->value().to_sstring() : sstring("-inf");
+        auto range_end = req.range.end() ? req.range.end()->value().to_sstring() : sstring("+inf");
+        co_await db::qctx->execute_cql(cql, req.table_uuid, req.repair_time, req.repair_uuid, req.keyspace_name, req.table_name, range_start, range_end).discard_result();
```

Comment:
We have to make sure either both or neither of the system table and the in-memory update succeed.

Reply:
It is fine only one of them is udpated. We can reply on the next repair to update the in-memory or next reboot. This will not cause correct issue.